acquire_token.go

```diff
@@ -115,7 +115,15 @@ func RegisterClient(namespace Namespace) (*config.PrefixEntry, error) {
 func AcquireToken(destination *url.URL, namespace Namespace, isWrite bool) (string, error) {
 	log.Debugln("Acquiring a token from configuration and OAuth2")
 
-	if len(namespace.Issuer) == 0 {
+	if namespace.CredentialGen == nil || namespace.CredentialGen.Strategy == nil {
+		return "", fmt.Errorf("Credential generation scheme unknown for prefix %s", namespace.Path)
+	}
+	if *namespace.CredentialGen.Strategy != "OAuth2" {
+		return "", fmt.Errorf("Unknown credential generation strategy (%s) for prefix %s",
```

Comment:
Slightly misleading: "Vault" isn't an unknown credential generation strategy, just an apparently unsupported one...

Reply:
Updated in latest commit to a switch statement with a better error message.